sysdeps/managarm/generic/file.cpp

```diff
@@ -3917,6 +3917,73 @@ int sys_pread(int fd, void *buf, size_t n, off_t off, ssize_t *bytes_read) {
 	}
 }
 
+int sys_pwrite(int fd, const void *buf, size_t n, off_t off, ssize_t *bytes_written) {
+	SignalGuard sguard;
+	HelAction actions[5];
+	globalQueue.trim();
+
+	auto handle = getHandleForFd(fd);
+	if (!handle)
+		return EBADF;
+
+	managarm::fs::CntRequest<MemoryAllocator> req(getSysdepsAllocator());
+	req.set_req_type(managarm::fs::CntReqType::PT_PWRITE);
+	req.set_fd(fd);
+	req.set_size(n);
+	req.set_offset(off);
+
+	frg::string<MemoryAllocator> ser(getSysdepsAllocator());
+	req.SerializeToString(&ser);
+	actions[0].type = kHelActionOffer;
+	actions[0].flags = kHelItemAncillary;
+	actions[1].type = kHelActionSendFromBuffer;
+	actions[1].flags = kHelItemChain;
+	actions[1].buffer = ser.data();
+	actions[1].length = ser.size();
+	actions[2].type = kHelActionImbueCredentials;
+	actions[2].flags = kHelItemChain;
+	actions[3].type = kHelActionSendFromBuffer;
+	actions[3].flags = kHelItemChain;
+	actions[3].buffer = const_cast<void *>(buf);
+	actions[3].length = n;
+	actions[4].type = kHelActionRecvInline;
+	actions[4].flags = 0;
+	HEL_CHECK(helSubmitAsync(handle, actions, 5,
+			globalQueue.getQueue(), 0, 0));
+
+	auto element = globalQueue.dequeueSingle();
+	auto offer = parseHandle(element);
+	auto send_req = parseSimple(element);
+	auto imbue_creds = parseSimple(element);
+	auto send_data = parseSimple(element);
+	auto recv_resp = parseInline(element);
+
+	HEL_CHECK(offer->error);
+	HEL_CHECK(send_req->error);
+	HEL_CHECK(imbue_creds->error);
+	HEL_CHECK(recv_resp->error);
+
+	managarm::fs::SvrResponse<MemoryAllocator> resp(getSysdepsAllocator());
+	resp.ParseFromArray(recv_resp->data, recv_resp->length);
+/*	if(resp.error() == managarm::fs::Errors::NO_SUCH_FD) {
+		return EBADF;
+	}else*/
```

Comment:
?

Reply:
Likely copy paste mistake, I'll investigate and remove it / uncomment it if it is used.